ethcore/src/snapshot/service.rs

```diff
@@ -481,15 +481,26 @@ impl Service {
 			let mut reader = self.reader.write();
 			*reader = None; // destroy the old reader if it existed.
 
+			let recovery = self.temp_recovery_dir();
 			let snapshot_dir = self.snapshot_dir();
 
 			if snapshot_dir.exists() {
+				for file in fs::read_dir(&snapshot_dir)? {
+					let file = file?;
+
+					let new_path = recovery.with_file_name(file.file_name());
+
+					fs::copy(file.path(), &new_path)?;
```

Comment:
copying the whole current snapshot every time we attempt to overwrite the old? also doesn't account for subdirectories.

I think it's fine if recovery failure ends up with the local snapshot also disappearing. We won't use nearly as much space on disk and it's such an edge case that it's not worth the extra cost.

Reply:
actually I think this code is completely broken. We only want the files in `temp_recovery_dir`, not the files in `snapshot_dir`. Copying them over will cause us to keep around all the snapshots we've ever downloaded (as opposed to the single most recent one).

and only the oldest of them will be accessible: one of the files in the current snapshot dir is a `MANIFEST`, which would then overwrite the `MANIFEST` in the temp recovery dir.

Reply:
Ok, we need to chat about this so I can get it right. I have no idea what the structure of this on the file system is, I was just trying to infer it from the code.

Reply:
once we've restored from a snapshot, we keep it on disk in the `snapshot_dir` and repropagate. but until the restoration is finished we keep the snapshot in a temporary directory. If there's an old snapshot, it's fine to overwrite because we would have only restored if the new one is more recent. The changes you included make it so we keep every snapshot we've ever downloaded, but only repropagate the oldest.